### ISSUE TITLE
netty4-http producer fails with RejectedExecutionException

### ISSUE THREAD

**Issue by hholtzhausen** [apache/camel-quarkus#132]
A netty4-http producer is started and stopped immediately during application init.  The route
then fails with a _java.util.concurrent.RejectedExecutionException_  when executed.

Producer debug logs during startup:
```
2019-08-13 15:51:38,827 DEBUG [org.apa.cam.com.net.htt.NettyHttpProducer] (main) Starting producer: Producer[http://localhost:8080/greeter/hello]
2019-08-13 15:51:38,869 DEBUG [org.apa.cam.com.net.htt.NettyHttpProducer] (main) Created NettyProducer pool[maxActive=-1, minIdle=0, maxIdle=100, minEvictableIdleTimeMillis=300000] -> org.apache.commons.pool.impl.GenericObjectPool@1e3e1014
2019-08-13 15:51:38,870 DEBUG [org.apa.cam.com.net.htt.NettyHttpProducer] (main) Stopping producer at address: localhost:8080
2019-08-13 15:51:38,885 DEBUG [org.apa.cam.com.net.htt.NettyHttpProducer] (main) Stopping producer with channel pool[active=0, idle=0]
2019-08-13 15:51:38,887 DEBUG [org.apa.cam.com.net.htt.NettyHttpProducer] (main) Stopping producer: Producer[http://localhost:8080/greeter/hello]
```

The exception can be reproduced using:
[https://github.com/hholtzhausen/examples/tree/master/camel-quarkus-netty4-http-producer](https://github.com/hholtzhausen/examples/tree/master/camel-quarkus-netty4-http-producer)

**Comment by lburgazzoli** [comment]
Can you remove @ApplicationScoped from the route ?

**Comment by hholtzhausen** [comment]
Same behaviour and exception without @ApplicationScoped annotation.
https://github.com/hholtzhausen/examples/commit/58e123d1852c4ea409a08f67aa35b052ecbe05f1

**Comment by lburgazzoli** [comment]
last thing the I'll test, can you change quarkus.camel.defer-init-phase from false to true ?

**Comment by hholtzhausen** [comment]
Tried that to :)
https://github.com/hholtzhausen/examples/commit/2816c9eec4bbfd8f55e6b6ce9335c6b89f7bb04a

**Comment by lburgazzoli** [comment]
@hholtzhausen this looks like an issue on our side, as confirmation: does it fails in both jvm and native mode ?

**Comment by ppalaga** [comment]
I can see it failing in JVM mode.

**Comment by lburgazzoli** [comment]
Seems related to different behaviour of default LRU cache vs LRU cache based on caffeine

**Comment by jamesnetherton** [comment]
Since you mentioned the LRU cache. I wonder if it's similar to what I observed in [CAMEL-13504](https://issues.apache.org/jira/browse/CAMEL-13504).

**Comment by lburgazzoli** [comment]
Yes it is

**Comment by hholtzhausen** [comment]
> @hholtzhausen this looks like an issue on our side, as confirmation: does it fails in both jvm and native mode ?

I was able to reproduce this in native mode a week or so ago using camel-quarkus 0.0.3-SNAPSHOT and quarkus 999-SNAPSHOT. I used those versions to get around https://github.com/apache/camel-quarkus/issues/80 and a _java.lang.NoClassDefFoundError: org/conscrypt/BufferAllocator_ during native compile.

With the current  camel-quarkus 0.1.1-SNAPSHOT version I am getting the _java.lang.NoClassDefFoundError: org/conscrypt/BufferAllocator_ exception again during native compile.  I suspect the NoClassDefFoundError is a problem in quarkus 0.20.0.
